firmware/thruna/FSM/Src/Io/Io_AcceleratorPedals.c

```diff
@@ -1,76 +1,139 @@
-#define PAPPS_COS_LAW_DENOMINATOR (2.0f * PAPPS_LEN_A * PAPPS_LEN_B)
-#define PAPPS_COS_LAW_COEFFICIENT (PAPPS_LEN_A * PAPPS_LEN_A + PAPPS_LEN_B * PAPPS_LEN_B) / PAPPS_COS_LAW_DENOMINATOR
-// Calculate the distance of the pedal from the voltage
-#define PAPPS_LEN_PER_VOLTAGE \
-    ((PAPPS_LENGTH_FULL_PRESSED_MM - PAPPS_LENGTH_UNPRESSED_MM) / (PAPPS_FULL_PRESSED_POT_V - PAPPS_UNPRESSED_POT_V))
-// PAPPS length offset in mm
-#define PAPPS_LEN_OFFSET_MM (227.6f)
-#define PAPPS_RAW_VOLTAGE_TO_LEN_MM(voltage) (voltage * PAPPS_LEN_PER_VOLTAGE + PAPPS_LEN_OFFSET_MM)
+#define PAPPS_COS_LAW_DENOMINATOR (2.0f * PAPPS_LEN_A * LEN_B)
+#define PAPPS_COS_LAW_COEFFICIENT ((PAPPS_LEN_A * PAPPS_LEN_A + LEN_B * LEN_B) / PAPPS_COS_LAW_DENOMINATOR)
+
+#define SAPPS_COS_LAW_DENOMINATOR (2.0f * SAPPS_LEN_A * LEN_B)
+#define SAPPS_COS_LAW_COEFFICIENT ((SAPPS_LEN_A * SAPPS_LEN_A + LEN_B * LEN_B) / SAPPS_COS_LAW_DENOMINATOR)
+
+// Macro to calculate the linear change in PAPPS length per change in voltage (we only use max because min = 0)
+#define APPS_VOLTAGE_PER_MM ((MAX_POT_LENGTH - MIN_POT_LENGTH) / MAX_POT_V)
+
+// linear equation to calculate current length using a rate of change from potientiometer
+// here max length here is the Yint and apps voltager per mm is the slope
+#define RAW_VOLTAGE_TO_LEN_MM(voltage) (MAX_POT_LENGTH - voltage * APPS_VOLTAGE_PER_MM)
+
+// calculating the unpressed and pressed PAPPS/SAPPS length based on the measured pot readings
+#define PAPPS_LENGTH_UNPRESSED_MM (RAW_VOLTAGE_TO_LEN_MM(PAPPS_UNPRESSED_POT_V))
+#define PAPPS_LENGTH_FULLY_PRESSED_MM (RAW_VOLTAGE_TO_LEN_MM(PAPPS_FULL_PRESSED_POT_V))
+
+#define SAPPS_LENGTH_UNPRESSED_MM (RAW_VOLTAGE_TO_LEN_MM(SAPPS_UNPRESSED_POT_V))
+#define SAPPS_LENGTH_FULLY_PRESSED_MM (RAW_VOLTAGE_TO_LEN_MM(SAPPS_FULL_PRESSED_POT_V))
+
+// OC & SC bounds
+#define PAPPS_MIN_V (PAPPS_UNPRESSED_POT_V - 0.5f)
+#define PAPPS_MAX_V (PAPPS_FULL_PRESSED_POT_V + 0.5f)
+
+#define SAPPS_MIN_V (SAPPS_UNPRESSED_POT_V - 0.5f)
+#define SAPPS_MAX_V (SAPPS_FULL_PRESSED_POT_V + 0.5f)
+
+#define DEAD_ZONE_PERCENT (10.0f)
+
+static float PAPPS_rest_angle;
+static float PAPPS_max_angle;
+
+static float SAPPS_rest_angle;
+static float SAPPS_max_angle;
+
+// max and min angle calculation for PAPPS/SAPPS
+void Io_AcceleratorPedals_Init(void)
+{
+    PAPPS_rest_angle =
+        (acosf(PAPPS_COS_LAW_COEFFICIENT - (powf(PAPPS_LENGTH_UNPRESSED_MM, 2) / PAPPS_COS_LAW_DENOMINATOR)));
+    PAPPS_max_angle =
+        PAPPS_rest_angle -
+        (acosf(PAPPS_COS_LAW_COEFFICIENT - (powf(PAPPS_LENGTH_FULLY_PRESSED_MM, 2) / PAPPS_COS_LAW_DENOMINATOR)));
 
-#define MIN_PEDAL_VOLTAGE (0.5f)
-#define MAX_PEDAL_VOLTAGE (3.1f)
+    SAPPS_rest_angle =
+        (acosf(SAPPS_COS_LAW_COEFFICIENT - (powf(SAPPS_LENGTH_UNPRESSED_MM, 2) / SAPPS_COS_LAW_DENOMINATOR)));
+    SAPPS_max_angle =
+        SAPPS_rest_angle -
+        (acosf(SAPPS_COS_LAW_COEFFICIENT - (powf(SAPPS_LENGTH_FULLY_PRESSED_MM, 2) / SAPPS_COS_LAW_DENOMINATOR)));
+}
```

Comment:
Use your help function for this, no need to copy paste something that we already have a helper for